library/Icingadb/Widget/ItemList/ObjectList.php

```diff
@@ -0,0 +1,241 @@
+<?php
+
+/* Icinga DB Web | (c) 2025 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Icingadb\Widget\ItemList;
+
+use Icinga\Exception\NotImplementedError;
+use Icinga\Module\Icingadb\Common\DetailActions;
+use Icinga\Module\Icingadb\Model\Comment;
+use Icinga\Module\Icingadb\Model\DependencyNode;
+use Icinga\Module\Icingadb\Model\Downtime;
+use Icinga\Module\Icingadb\Model\History;
+use Icinga\Module\Icingadb\Model\Host;
+use Icinga\Module\Icingadb\Model\NotificationHistory;
+use Icinga\Module\Icingadb\Model\RedundancyGroup;
+use Icinga\Module\Icingadb\Model\Service;
+use Icinga\Module\Icingadb\Model\UnreachableParent;
+use Icinga\Module\Icingadb\Model\User;
+use Icinga\Module\Icingadb\Model\Usergroup;
+use Icinga\Module\Icingadb\Redis\VolatileStateResults;
+use Icinga\Module\Icingadb\View\CommentRenderer;
+use Icinga\Module\Icingadb\View\DowntimeRenderer;
+use Icinga\Module\Icingadb\View\HostRenderer;
+use Icinga\Module\Icingadb\View\RedundancyGroupRenderer;
+use Icinga\Module\Icingadb\View\ServiceRenderer;
+use Icinga\Module\Icingadb\View\UsergroupRenderer;
+use Icinga\Module\Icingadb\View\UserRenderer;
+use Icinga\Module\Icingadb\Widget\Notice;
+use InvalidArgumentException;
+use ipl\Html\HtmlDocument;
+use ipl\I18n\Translation;
+use ipl\Orm\Model;
+use ipl\Stdlib\Filter;
+use ipl\Web\Layout\DetailedItemLayout;
+use ipl\Web\Layout\ItemLayout;
+use ipl\Web\Layout\MinimalItemLayout;
+use ipl\Web\Widget\ItemList;
+use ipl\Web\Widget\ListItem;
+
+/**
+ * ObjectList
+ *
+ * Create a list of icingadb objects
+ *
+ * @template Item of RedundancyGroup|Service|Host|Usergroup|User|Comment|Downtime
+ *
+ * @extends ItemList<Item>
```

Comment:
I've just updated https://github.com/Icinga/ipl-web/pull/252 again, so that `ItemList` differentiates between what kind of data is provided to the constructor and what createListItem returns. To facilitate this, use this template definition:

```suggestion
 * @template Result of DependencyNode|Service|Host|Usergroup|User|Comment|Downtime
 * @template Item of RedundancyGroup|Service|Host|Usergroup|User|Comment|Downtime
 * @extends ItemList<Result, Item>
```

I've left out `UnreachableParent` because it's a `DependencyNode` anyway.

Afterwards, you can drop the documentation of createListItem here.

Reply:
Updated the suggestion